src/figuresClasses.ts

```diff
@@ -1,11 +1,86 @@
-export interface Figure {}
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: string;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public sideA: number,
+    public sideB: number,
+    public sideC: number,
+  ) {
+    if (
+      sideA >= sideB + sideC ||
+      sideB >= sideA + sideC ||
+      sideC >= sideA + sideB
+    ) {
+      throw new Error(
+        'The sum of the lengths of any two sides ' +
+          'must be greater than the length of the remaining side.',
+      );
+    }
+
+    if (sideA <= 0 || sideB <= 0 || sideC <= 0) {
+      throw new Error('All sides must be greater than 0.');
+    }
+  }
+
+  getArea(): number {
```

Comment:
The `getArea` method should not accept any parameters as per the checklist. All necessary values should be taken directly from the object's fields, which you are already doing, so the method signature should be `getArea(): number` without parameters.